### ISSUE TITLE
Id like to add some features but am not sure where to start.

### ISSUE THREAD

**Issue by mikeder** [devsnd/cherrymusic#500]
Hello and first of all thank you for creating this excellent music server! I thoroughly enjoy using this bit of software, as do my friends. I apologize if this isn't the right place to make a request like this so please pardon my approach.

Id like to a add a few playlist features but I am not sure which files to begin to edit. Maybe you guys don't have time to work on these yourself but if you don't mind pointing me in the right direction I can give it a shot. Here's what I have in mind:
1. Ability to 'queue' next song for playback by pressing 'q' while window is in focus or from the drop-down arrow next to the track name.
2. Add a song from a current playlist to a new one via drop-down arrow.
3. Once meta data has been loaded from files it should be stored in the database so it doesn't have to be read from the file every time.
4. Single click to select song, double to play/add to playlist. (This would be required for the queue feature.)

Thank you for you time and consideration, I look forward to contributing!

**Comment by tilboerner** [comment]
> Introducing a new Database table can be done by creating a new module in cherrymusicserver/database/defs, see the useroptions module for example.

Yeah, actually we created several databases (sqlite3 files) for the different major components, `useroptions.db` being one of them.

Each subdirectory in `cherrymusicserver/database/defs` contains the (versioned) definition of a database. Each version is a numbered subdirectory that holds SQL scripts to *create*, *update* (from the previous version) or *drop* the database. Aptly named *after.sql* :bowtie: is run after creation or update and can be used to create indices or triggers.

To **create a database**, create a subdirectory in `defs`. Its name will become the basename of the .db file. Create a subdirectory in it called `1`. (`0` is for versions older than the versioning scheme.) Create at least `create.sql` and `drop.sql` in it. The database will be created automatically the next time you start cherrymusic.

```sh
$ cd cherrymusicserver/database/defs
$ mkdir -p DATABASE_NAME/1 && touch DATABASE_NAME/1/{create,drop}.sql
```

To **amend a database**, e.g., to add a table, create a new version in its `defs` subdirectory. Create at least `create.sql`, `update.sql` and `drop.sql` in it. The database will be updated automatically the next time cherrymusic is started. If you include a file called `prompt`, the admin will be asked for consent before the update and be given the reason contained in this file. Do this if you touch existing tables during the update.

To run some basic **tests for your sql scripts**, execute

    $ ./runtests database

in the cherrymusic repository.

See `defs/cherry.cache/1` for an example definition, and see class `PlaylistDB` in `playlistdb.py` for an example of how to connect to a cherrymusic database.

If you have further questions, please let us know.
